File: ticket-backend/src/main/java/aics/domain/hall/dtos/HallListItemDto.java
package aics.domain.hall.dtos;

import aics.domain.hall.entities.Hall;
import aics.infrastructure.core.LabelValue;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@Accessors(chain = true)
public class HallListItemDto implements Serializable {
    private Long hallId;
    private String name;
    private String description;
    private int seatsRows;
    private int seatsColumns;
    private LabelValue<Long> providerRef;

    public static HallListItemDto fromHall(Hall hall) {
        if (hall == null) {
            return null;
        }
        LabelValue<Long> providerRef = hall.getProvider() != null
            ? new LabelValue<>(hall.getProvider().getName(), hall.getProvider().getProviderId())
            : null;
        return new HallListItemDto()
            .setHallId(hall.getHallId())
            .setName(hall.getName())
            .setDescription(hall.getDescription())
            .setSeatsRows(hall.getSeatsRows())
            .setSeatsColumns(hall.getSeatsColumns())
            .setProviderRef(providerRef);
    }
}

File: ticket-frontend/src/modules/hall/hall-list-item-dto.ts
import { LabelValue } from '../core/label-value';
import { TypeUtils } from '../core/type-utils';

export class HallListItemDto {
    hallId: number = 0;
    name: string = '';
    seatsRows: number = 0;
    seatsColumns: number = 0;
    description: string = '';
    providerRef: LabelValue<number> | null = null;

    static fromObj(obj: any): HallListItemDto | null {
        if (!obj) {
            return null;
        }
        const providerDto: HallListItemDto = new HallListItemDto();
        providerDto.hallId = obj.hallId;
        providerDto.name = obj.name;
        providerDto.seatsRows = obj.seatsRows;
        providerDto.seatsColumns = obj.seatsColumns;
        providerDto.description = obj.description;
        providerDto.providerRef = LabelValue.fromObj(obj.providerRef);
        return providerDto;
    }

    static listFromObjList(objs: any[]): HallListItemDto[] {
        if (!objs) {
            return [];
        }
        return objs.map(HallListItemDto.fromObj).filter(TypeUtils.isNonNullable);
    }
}